### DISCUSSION TITLE
Step-by-step guide to add HomeAssistant Integration and get local MQTT

### DISCUSSION THREAD

**Discussion by CWILDMOUNTAIN** [reinhard-brandstaedter/solarflow-control#261] [Show and tell]

**Comment by devninja-sudo** [comment]
## Step 1: Verify the Mosquitto MQTT Broker

1. **Ensure Mosquitto is Running**:
   - Go to **Settings > Add-ons, Backups & Supervisor > Dashboard** in Home Assistant.
   - Ensure the **Mosquitto broker** add-on is installed and running. If not, install it from the Add-on Store and start it.

2. **Check Mosquitto Configuration**:
   - Click on the Mosquitto broker add-on.
   - Review the **Configuration** tab to ensure it is set up correctly (allow_anonymous true).
## Step 2: Take your hub offline (strongly recommended)
   - Have a look at https://github.com/reinhard-brandstaedter/solarflow-bt-manager
## Step 3: Download and Run the Solarflow Control Docker Image

1. **Open Terminal**:
   - Use Terminal or an SSH client to access your environment if necessary.

2. **Pull the Docker Image**:
   - Run the following command to pull the latest Solarflow Control image:
     ```sh
     docker pull rbrandstaedter/solarflow-control:latest
     ```

3. **Create Configuration Directory and File**:
   - Create the `config.ini` file in the solarflow directory with the following content:
     ```ini
      [global]
      # DTY Type: either OpenDTU or AhoyDTU
      dtu_type = OpenDTU
      # Smartmeter Type: either Smartmeter (generic, Tasmota, Hichi, ...), PowerOpti, Shelly3EM
      smartmeter_type = Smartmeter
      
      # Geolocation LAT/LNG
      #latitude =
      #longitude =
      
      # Offset in minutes after sunrise/before sunset. Can be used to set the duration of what is considered "night"
      sunrise_offset = 0
      sunset_offset = 0
      
      [solarflow]
      # The product ID specifies the model of Solarflow hub to use: Hub-1200: "73bkTV" Hub-2000: "A8yh63"
      # defaults to 73bkTV
      product_id = A8yh63
      # The device ID of your Solarflow Hub (typically 8 characters), you can get these either with solarflow-bt-manager or the solarflow-statuspage
      device_id = xxxxxx
      # The time interval in hours that solarflow-control will try to ensure a full battery
      # (i.e. no discharging if battry hasn't been at 100% for this long)
      full_charge_interval = 576
      control_bypass = true
      [mqtt]
      # Your local MQTT host configuration
      mqtt_host = 192.168.178.21
      #mqtt_port =
      #mqtt_user =
      #mqtt_pwd =
      
      [opendtu]
      # The MQTT base topic your OpenDTU reports to (as configured in OpenDTU UI)
      base_topic = solar
      # your Inverters serial number
      inverter_serial = 11419xxxxxxx
      
      # List of indices of the inverter channels/ports (as reported in the DTU) that the Solarflow Hub is connected to
      # typically the index starts at 1 as 0 is the output channel of the inverter
      # e.g. 1,3 or 3 or [1,3]
      sf_inverter_channels = [1,2]
      
      [smartmeter]
      base_topic = shellypro3em/status/em:0
      # if the published value at the base_topic is a JSON type then these accessors are used to get the power values
      # e.g. if Smartmeter reader posts { "Power": {"Power_curr": 120, "Total_in": 12345.6} }
      cur_accessor = total_act_power
      total_accessor = total_aprt_power
      rapid_change_diff = 100
      [poweropti]
      # Username and password for you Powerfox API to get readings (internet connection required)
      poweropti_user = <PowerFox API user>
      poweropti_password = <Powerfox API password>
      
      [shelly3em]
      # The MQTT base topic your Shelly 3EM (Pro) is posting it's telemetry data to
      # Note: you have to configure your Shelly to use MQTT
      base_topic = shellies/shellyem3/
      
      [control]
      min_charge_power = 0
      max_discharge_power = 800
      max_inverter_limit = 800
      limit_inverter = false
      inverter_min_limit = 100
      BATTERY_LOW=15
     ```

4. **Run the Docker Container with Mounted Config File**:
   - Run the following command to start the Docker container, mounting the `config.ini` file:
     ```sh
     docker run -v ${PWD}/src/config.ini:/solarflow/config.ini --name solarflow-control rbrandstaedter/solarflow-control:latest
     ```


## Troubleshooting

- **Check Docker Logs**:
  - If the Solarflow Control container is not working as expected, check the logs for any errors:
    ```sh
    docker logs solarflow-control
    ```

**Reply by Armageddit** [reply]
Is it possible to create a Homeassistant addon?
I've been sitting at it for hours and I can't start either the solarflow-bt-manager or the solarflow-control.

There are instructions everywhere that I can't get any further due to errors :(
there is no docker command.

I have homeassistantOS running on a Raspberry Pi 4.
If I have to keep my 200w guzzling PC running to manage the battery, I can sell the battery straight away.

**Reply by reinhard-brandstaedter** [reply]
You have a few options:
Install docker (or similar container image) on your Raspi.
HA addons are also docker containers btw. (check out [this discussion](https://github.com/reinhard-brandstaedter/solarflow-control/issues/204)

Bt-manager is non-containerized as it is only required to run once and it needs direct access to the bluetooth stack.
sf-control can also be run without docker, by directly executing the python script, but I really encourage to use it via docker to avoid any software dependecies.
